Data/NF/Internal.hs

```diff
@@ -51,5 +61,23 @@ newtype NF a
 type role NF nominal
 #endif
 
+-- UGH! GHC swapped the constraint order between
+-- 7.10 and 8.0. If anyone's using some beta compiler,
+-- it might throw up.
+--
+-- Very unfortunately, GHC (as of 8.0) does not allow the pattern
+-- synonym to be *matched* without an NFData context. Hopefully
+-- this will be fixed eventually.
```

Comment:
I guess I didn't make a ticket before, but now I have:
https://ghc.haskell.org/trac/ghc/ticket/12426

On Jul 22, 2016 6:14 PM, "Edward Z. Yang" <notifications@github.com> wrote:

> In Data/NF/Internal.hs
> <https://github.com/ezyang/nf/pull/6#discussion_r71953725>:
>
> > @@ -51,5 +61,23 @@ newtype NF a
> >  type role NF nominal
> >  #endif
> >
> > +-- UGH! GHC swapped the constraint order between
> > +-- 7.10 and 8.0. If anyone's using some beta compiler,
> > +-- it might throw up.
> > +--
> > +-- Very unfortunately, GHC (as of 8.0) does not allow the pattern
> > +-- synonym to be *matched* without an NFData context. Hopefully
> > +-- this will be fixed eventually.
>
> Is there a GHC bug for it?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ezyang/nf/pull/6/files/db57fa061540b46fb54928e8131e84be376b7fe9#r71953725>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABzi_TA8a6SM-i2ugC6epjeaD8HCfqOBks5qYUDdgaJpZM4JTKLc>
> .
>